has-selector.js

```diff
@@ -3,65 +3,81 @@
             x = new XMLHttpRequest();
             x.addEventListener("load", addAliases.bind(null, x, n[i]));
-            x.open("GET", n[i].getAttribute("href"), true);
+            x.open("GET", n[i].getAttribute("href"), false);
             x.send();
         }
+
+        toggleAliases();
+    }
+
+    // This function parses each <style> element's stylesheet.
+    // Without it, we'd have no way of finding the :has selectors within <style> elements.
+    function parseInlineStylesheets() {
+        var s = "style[type='text/css']:not(:empty)"; // Selector matching <style> elements
+        var n = document.querySelectorAll(s);         // list of matched <style> elements
+        var i;                                        // index of the current <style> element
+
+        for (i = 0; i < n.length; i++) {
+            addAliases({responseText: n[i].textContent}, n[i]);
+        }
     }
 
     // This function finds the :has selectors, and replaces each one with a class alias.
     // It also replaces each <link> element with a <style> element containing the aliases.
     // Without it, we're unable to insert the aliases into the CSS in the appropriate position.
-    function addAliases(xhr, link) {
-        var s = document.createElement("style"); // <style> element to replace the <link> element
-        var r = /[)(}{@;,]|:has\b/g;             // RegExp to find important symbols within the CSS
-        var t = xhr.responseText;                // the CSS text loaded via AJAX
-        var inHasDec = 0;                        // indicates whether we're in a :has declaration
-        var inAtSlct = 0;                        // indicates whether we're in an at-rule selector
-        var inAtBlck = 0;                        // indicates whether we're in an at-rule block
-        var inParens = 0;                        // indicates whether we're within parentheses
-        var inStyles = 0;                        // indicates whether we're within a style block
-        var foundAny;                            // indicates whether any :has declarations were found
-        var i = 0;                               // index at which the current selector begins
-        var h = 0;                               // index at which the :has declaration begins
-        var f;                                   // the results of RegExp search
-
-        while (f = r.exec(t)) switch (f[0]) {
+    function addAliases(xhr, oldElem) {
+        var newStyle = document.createElement("style"); // <style> element to replace the <link> element
+        var r = /[)(}{@;:>+~, ]|\bhas\b/g;              // RegExp to find important symbols within the CSS
+        var styleTxt = xhr.responseText;                // the CSS text loaded via AJAX
+        var inHasDec = 0;                               // indicates whether we're in a :has declaration
+        var inPrnths = 0;                               // indicates whether we're in parentheses
```

Comment:
I think you had it better before this revision, "parens" is a very idiomatic abbreviation for parentheses ;-)